emukit/core/loop/outer_loop.py

```diff
@@ -54,18 +54,22 @@ def __init__(self, candidate_point_calculator: CandidatePointCalculator,
         self.loop_start_event = EventHandler()
         self.iteration_end_event = EventHandler()
 
-    def run_loop(self, user_function: Union[UserFunction, Callable], stopping_condition: Union[StoppingCondition, int],
-                 context: dict=None) -> None:
+    def run_loop(self, user_function: Union[UserFunction, Callable],
+                 stopping_condition: Union[StoppingCondition, int, List[StoppingCondition]],
+                 context: dict = None) -> None:
         """
         :param user_function: The function that we are emulating
-        :param stopping_condition: If integer - a number of iterations to run, if object - a stopping condition object
-                                   that decides whether we should stop collecting more points
+        :param stopping_condition: If integer - a number of iterations to run, if object (or list of such objects) - a stopping
+                        condition object that decides whether we should stop collecting more points
         :param context: The context is used to force certain parameters of the inputs to the function of interest to
                         have a given value. It is a dictionary whose keys are the parameter names to fix and the values
                         are the values to fix the parameters to.
         """
-        if not (isinstance(stopping_condition, int) or isinstance(stopping_condition, StoppingCondition)):
-            raise ValueError("Expected stopping_condition to be an int or a StoppingCondition instance, "
+
+        if not ((isinstance(stopping_condition, list) and
+                all([isinstance(condition, StoppingCondition) for condition in stopping_condition])) or
+                isinstance(stopping_condition, int) or isinstance(stopping_condition, StoppingCondition)):
```

Comment:
This `if` is very hard to parse now! Can you break it down somehow? One idea is to create boolean variables for each condition. Something like that:
```
   is_int = isinstance(stopping_condition, int)
   is_single_condition = ...
   is_condition_list = ...
```
Don't have to be like that, the point is to make this `if` readable.

Reply:
@apaleyes might it we worth doing a "OR" and "AND" stopping condition that handles multiple stopping conditions. So you could have multiple stopping conditions that go into the OrStoppingCondition for example. Then it would be easier to handle the inputs here.

Reply:
@mmahsereci in the future maybe, and there might be some nice ways to implement that. but for now that seems overkill. let's deal with it once someone actually needs it

Reply:
after offline chat with @mmahsereci we've decided to cut an issue to do something more flexible here: #273 . @henrymoss please ignore it for now, it is certainly outside the scope of this PR